src/components/game-over/game-over.tsx

```diff
@@ -0,0 +1,48 @@
+import { useSignal } from "@preact/signals";
+import { CardComponent } from "@components/shared/card/card";
+import { KeyValueComponent } from "@components/shared/key-value-list/key-value";
+import { KeyValueListComponent } from "@components/shared/key-value-list/key-value-list";
+import { getRandomCongratulation } from "@utils/text.utils";
+import { useContext, useState } from "preact/compat";
+import { GameStateContext, StatisticContext } from "../../interfaces/context";
+import "./game-over.css";
+import { ModalComponent } from "@components/shared/modal/modal";
+
+export function GameOverComponent() {
+  const { timeSpentInSeconds, cardsFlipCount } = useContext(GameStateContext);
+  const { averageCardFlipsCount, averageTimeSpentInSeconds } =
+    useContext(StatisticContext);
+
+  const [headerMessage] = useState(getRandomCongratulation());
```

Comment:
It seems that this message might just be plain text; no reactive state is necessary.